crates/common/mqtt_channel/src/connection.rs

```diff
@@ -186,16 +186,40 @@ impl Connection {
                 Ok(Event::Incoming(Packet::ConnAck(ack))) => {
                     if let Some(err) = MqttError::maybe_connection_error(&ack) {
                         eprintln!("ERROR: Connection Error {}", err);
-                    }
-                    // Workaround for  https://github.com/bytebeamio/rumqtt/issues/250
-                    // If session_name is not provided, then re-subscribe
-                    else if config.session_name.is_none() {
-                        let subscriptions = config.subscriptions.filters();
-                        // Need check here otherwise it will hang waiting for a SubAck, and none will come when there is no subscription.
-                        if subscriptions.is_empty() {
-                            break;
+                    } else {
+                        match config.session_name {
+                            Some(ref sname) => {
+                                if config.initial_message {
+                                    println!(
+                                        "........Connected.........................{:?}",
+                                        sname
+                                    );
+
+                                    let health_status = r#"{"status": "up","type": "service"}"#;
+
+                                    let _ = mqtt_client
+                                        .publish(
+                                            format!("tedge/health/{sname}"),
+                                            rumqttc::QoS::AtLeastOnce,
+                                            true,
+                                            health_status.as_bytes().to_vec(),
+                                        )
+                                        .await;
```

Comment:
This message is thin-edge specific while this crate should not. The message must not be forged by the library but provided by the user of the library.